src/analysis/retail/paladin/retribution/normalizers/DivineHammerNormalizer.tsx

```diff
@@ -0,0 +1,52 @@
+import SPELLS from 'common/SPELLS';
+import { EventType, CastEvent } from 'parser/core/Events';
+import EventsNormalizer from 'parser/core/EventsNormalizer';
+
+const DIVINE_HAMMER_COOLDOWN_MS = 120000;
+
+// Pressing Divine Hammer fires multiple Cast Success events which messes with the cooldown graph (or what it depends on)
+// This Normalizer basically marks any "fake" casts as Free Cast
+class DivineHammerNormalizer extends EventsNormalizer {
+  normalize(events: any[]): any[] {
+    const fixedEvents: any[] = [];
```

Comment:
these should be `AnyEvent` instead of `any`:

```suggestion
  normalize(events: AnyEvent[]): AnyEvent[] {
    const fixedEvents: AnyEvent[] = [];
```

Reply:
should I also type `fabricatedEvent` as `AnyEvent` and change  `_fabricated` to `__modified` (or `__fabricated` maybe, i was missing an underscore, though `__modified` feels more accurate) for this to work ?

Reply:
ooh, good catch. yes, `__modified` is more appropriate here